src/Orleans/Runtime/DedicatedThreadPool.cs

```diff
@@ -0,0 +1,756 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="Helios.Concurrency.DedicatedThreadPool.cs" company="Akka.NET Project">
+//     Copyright (C) 2009-2016 Lightbend Inc. <http://www.lightbend.com>
+//     Copyright (C) 2013-2016 Akka.NET project <https://github.com/akkadotnet/akka.net>
+// </copyright>
+//-----------------------------------------------------------------------
+
+/*
+ * Copyright 2015 Roger Alsing, Aaron Stannard
+ * Helios.DedicatedThreadPool - https://github.com/helios-io/DedicatedThreadPool
```

Comment:
Are there any Orleasns-specific changes in this code?

This `DedicatedThreadPool ` class is available as a NuGet package, which I suggest is the easiest and fastest way to do this reuse.
https://www.nuget.org/packages/Helios.DedicatedThreadPool/

@sergeybykov - FWIW, the original Akka.Net code is Apache-2.0 licensed, so if we really want to copy-paste include this Apache-2 licensed code into the Orleans (MIT) code base then we probably need to get some experts to confirm everything is OK? IANAL and i don't recall what the prevailing advice is about cross-/re-licensing of external OSS code ;)
https://github.com/helios-io/DedicatedThreadPool

It would be so much easier to re-use the NuGet package! ;)